src/components/board/Board.jsx

```diff
@@ -6,42 +6,64 @@ import { useCallback, useEffect, useRef } from 'react';
 import { useSelector } from 'react-redux';
 import { useParams } from 'react-router';
 import { useDraggable } from 'react-use-draggable-scroll';
-import { loadBoard } from '../../store/board/board.actions.js';
+import { loadBoard, updateBoard } from '../../store/board/board.actions.js';
 import { BoardHeader } from './components/BoardHeader.jsx';
 import { Column } from './components/Column.jsx';
 
 export function Board() {
   const { boardId } = useParams();
   const board = useSelector((state) => state.boardModule.board);
+  console.log('Board data:', board);
   const listRef = useRef();
   const { event } = useDraggable(listRef);
+  
+  useEffect(() => {
+    loadBoard(boardId);
+  }, [boardId]);
+
 
-  const moveTask = useCallback(
-    ({
-      taskIndexInStartGroup,
-      sourceGroupId,
-      destinationGroupId,
-      itemIndexInFinishGroup,
-    }) => {
-      const startGroupData = board.groups[sourceGroupId];
-      const destinationGroupData = board.groups[destinationGroupId];
-      const taskToMove = startGroupData.tasksIds[taskIndexInStartGroup];
-      const newStartGroupData = {
-        ...startGroupData,
-        tasksIds: startGroupData.tasksIds.filter((task) => task !== taskToMove),
-      };
-      const newDestinationTasks = destinationGroupData.tasksIds;
-      const newIndexInDestination = itemIndexInFinishGroup ?? 0;
-
-      newDestinationTasks.splice(newIndexInDestination, 0, taskToMove);
-
-      const newFinishGroupData = {
-        ...destinationGroupData,
-        tasksIds: newDestinationTasks,
-      };
-
-      console.log('moveTask newStartGroupData ', newStartGroupData);
-      console.log('moveTask newFinishGroupData ', newFinishGroupData);
+ const moveTask = useCallback(
+   ({
+     taskIndexInStartGroup,
+     sourceGroupId,
+     destinationGroupId,
+     itemIndexInFinishGroup,
+   }) => {
+     if (!board) return;
+
+     const updatedBoard = JSON.parse(JSON.stringify(board)); // Deep clone
```

Comment:
why???

Reply:
when you change nested states it is better to use deep clone (groups, tasks..) . this was the issue when we tried to render the changes after dnd actions, and didn't see any change. shallow clone like ...board will not work beacuse we need to create new objects for these nested states/properties..
i am not sure this is the optimised way to make a deep clone but we can ask yonatan about it.